scripts/zones/Halvung/npcs/_1qp.lua

```diff
@@ -0,0 +1,37 @@
+-----------------------------------
+-- Area: Halvung
+--  NPC: Decorative Bronze Gate (_1qp)
+-----------------------------------
+local ID = require("scripts/zones/Halvung/IDs")
+require("scripts/globals/npc_util")
+require("scripts/globals/items")
+require("scripts/globals/status")
+
+-----------------------------------
+local entity = {}
+
+entity.onTrade = function(player, npc, trade)
+    if npcUtil.tradeHas(trade, xi.items.HALVUNG_SHAKUDO_KEY) and
```

Comment:
```lua
    if
     npcUtil.tradeHas(trade, xi.items.HALVUNG_SHAKUDO_KEY) and
     npcUtil.tradeHas(trade, xi.items.HALVUNG_BRONZE_KEY) and
     npcUtil.tradeHas(trade, xi.items.HALVUNG_BRASS_KEY)
    then
        player:confirmTrade()
        npc:openDoor()
        player:messageSpecial(ID.text.KEY_BREAKS, 2221,2222,2223)
    end
```
per the style guide
    ```

Reply:
npcUtil.tradeHasExactly(trade, {item1, item2, item3})

Reply:
what he said and then you don't have to break a wide line in the 1st place. I'm still not used to these utils _existing_

..old grandpa Teo had to explicitly ask whats was in the trade boxes back in his day.

Reply:
> npcUtil.tradeHasExactly(trade, {item1, item2, item3})

it seems this removes the ability to use the items.lua?
    if npcUtil.tradeHasExactly(trade, {2221, 2222, 2223}) then
vice     if npcUtil.tradeHasExactly(trade, {HALVUNG_SHAKUDO_KEY, HALVUNG_BRONZE_KEY, HALVUNG_BRASS_KEY}) then

Reply:
`if npcUtil.tradeHasExactly(trade, {xi.items.HALVUNG_SHAKUDO_KEY, xi.items.HALVUNG_BRONZE_KEY, xi.items.HALVUNG_BRASS_KEY}) then`

Also I now see its still a bit wide.

Reply:
It wouldn't trade with the latter example, pushed with 2221,2222,2223 as that did work when testing it.

Reply:
you were missing `xi.items.` in your post though. I -know- it works, so check again you had to make a mistake

Reply:
the human readable form also works with the `messageSpecial()` call below it

Reply:
That makes sense.